Mdx/4 10  javax jakarta change

javac9 opened the pull request.

Commit:
[MDX-4/10] Ensure jakarta 8 is available during the transition

Commit:
[MDX-4/10][core] Use manifold macro conditionals to enable parallel usage of javax and jakarta imports

Commit:
[MDX-4/10][security] Use manifold macro conditionals to enable parallel usage of javax and jakarta imports

Commit:
manifold test modules

Commit:
[MDX-4/10][portal] Use manifold macro conditionals to enable parallel usage of javax and jakarta imports

Commit:
[MDX-4/10] Comment out some apparently unnecessary @Override annotations, other smaller refactorings

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/jmi/spi/RefRootPackage_1.java:
```diff
@@ -1846,7 +1854,7 @@ public void makeTransientAll(Collection pcs, boolean useFetchPlan) {
          * @see javax.jdo.PersistenceManager#makeTransientAll(java.lang.Object[], boolean)
          */
-        @Override
+//        @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/jmi/spi/Jmi1DataStoreCache.java:
```diff
@@ -120,7 +120,7 @@ public void evictAll(Collection xris) {
      */
     @SuppressWarnings("rawtypes")
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/jmi/spi/Jmi1DataStoreCache.java:
```diff
@@ -170,7 +170,7 @@ public void pinAll(Object... xris) {
      */
     @SuppressWarnings("rawtypes")
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/jmi/spi/Jmi1DataStoreCache.java:
```diff
@@ -220,7 +220,7 @@ public void unpinAll(Object... xris) {
      */
     @SuppressWarnings("rawtypes")
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/rest/DataObjectManager_1.java:
```diff
@@ -1341,7 +1348,7 @@ public Collection getObjectsById(
      * @see javax.jdo.PersistenceManager#getObjectsById(java.lang.Object[], boolean)
      */
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/rest/DataObjectManager_1.java:
```diff
@@ -1633,7 +1640,7 @@ public void makeTransientAll(Collection pcs) {
      * @see javax.jdo.PersistenceManager#makeTransientAll(java.lang.Object[], boolean)
      */
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/rest/DataObjectManager_1.java:
```diff
@@ -1987,7 +1994,7 @@ public void retrieveAll(
      * @see javax.jdo.PersistenceManager#retrieveAll(java.lang.Object[], boolean)
      */
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/view/ViewManager_1.java:
```diff
@@ -1044,7 +1044,7 @@ public Collection getObjectsById(
      * @see javax.jdo.PersistenceManager#getObjectsById(java.lang.Object[], boolean)
      */
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/view/ViewManager_1.java:
```diff
@@ -1248,7 +1248,7 @@ public void makeTransientAll(Collection pcs) {
      * @see javax.jdo.PersistenceManager#makeTransientAll(java.lang.Object[], boolean)
      */
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/accessor/view/ViewManager_1.java:
```diff
@@ -1578,7 +1578,7 @@ public void retrieveAll(
      * @see javax.jdo.PersistenceManager#retrieveAll(java.lang.Object[], boolean)
      */
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/caching/datastore/CacheAdapter.java:
```diff
@@ -148,7 +148,7 @@ public interface CacheAdapter extends JDODataStoreCache {
      */
     @SuppressWarnings("rawtypes")
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/caching/datastore/CacheAdapter.java:
```diff
@@ -193,7 +193,7 @@ default void evictAll (boolean subclasses, Class pcClass) {
      */
     @SuppressWarnings("rawtypes")
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/caching/datastore/CacheAdapter.java:
```diff
@@ -243,7 +243,7 @@ default void pinAll(
      */
     @SuppressWarnings("rawtypes")
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/persistence/spi/AbstractPersistenceManager.java:
```diff
@@ -567,7 +567,7 @@ public Object[] getObjectsById(Object... oids) {
      * @see javax.jdo.PersistenceManager#getObjectsById(java.lang.Object[], boolean)
      */
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY commented on core/src/main/java/org/openmdx/base/persistence/spi/AbstractPersistenceManager.java:
```diff
@@ -615,7 +615,7 @@ public void retrieveAll(Object... pcs) {
      * @see javax.jdo.PersistenceManager#retrieveAll(java.lang.Object[], boolean)
      */
-    @Override
+//    @Override
```
Why no more overriding?

DlRTY-HARRY reviewed (approved):
Provided the overrides will reappear…

Commit:
[MDX-4/10] Add deprecation/removal detail info to javadoc of methods from which @Override annotation was removed

Commit:
[MDX-4/10] Add close() method due to compiler complaint that it is expected there

Commit:
[MDX-4/10] Throw a runtime exception with the IllegalStateException as a parameter

Commit:
[MDX-4/10] Use manifold macro conditionals in remaining places to enable parallel usage of javax and jakarta imports

Commit:
[MDX-4/10] Add rt.jar to support usage of appropriate @Generated annotations in generated code

Commit:
[MDX-4/10] Make sure "portal" and "publish" modules get only included if flavour is not 4

Commit:
[MDX-4/10] Add newly required methods with UnsupportedOperationException as implementation thereof

Commit:
[MDX-4/10] Add newly required methods with UnsupportedOperationException as implementation thereof

Commit:
[MDX-4/10] Add @Deprecated annotation; remove unnecessary public visibility modifier from an interface

Commit:
[MDX-4/10][portal - REVERT] Use manifold macro conditionals to enable parallel usage of javax and jakarta imports

Commit:
[MDX-4/10] Fix generation of the @Generated annotation to support both javax and jakarta namespaces

Commit:
MDX-3 change The openMDX 4 JAR shall be able to build all flavours

Commit:
MDX-3 Mulzi-flavour support

- Use Jakarta Servket API
- Use flavour specific build directory in tests
- Exclude methods requiring JDO 3.2 classes

Commit:
MDX-3 Multi-flavour support

- Use Jakarta Servket API
- Use flavour specific build directory in tests
- Exclude methods requiring JDO 3.2 classes
- Ignore Eclipse & macOS Files

Commit:
MDX-4 fix Bereininge Tests

Commit:
MDX-4 fix Jakarta 8 & 10 support amended

- Missing manifold directives added
- Atomikos jakarta classifier added

Commit:
MDX-4 add support flavour specific local openmdx-base archive for
bootstrapping
